04.ls/ls.rb

```diff
@@ -2,26 +2,132 @@
 # frozen_string_literal: true
 
 require 'optparse'
+require 'etc'
 
 COLUMNS = 3
 TAB_WIDTH = 8
 
+ENTRY_TYPES = {
+  file: '-',
+  blockSpecial: 'b',
+  characterSpecial: 'c',
+  directory: 'd',
+  link: 'l',
+  fifo: 'p',
+  socket: 's',
+  unknown: 'w'
+}.freeze
+
+FILE_PERMISSIONS = {
+  '0' => '---',
+  '1' => '--x',
+  '2' => '-w-',
+  '3' => '-wx',
+  '4' => 'r--',
+  '5' => 'r-x',
+  '6' => 'rw-',
+  '7' => 'rwx'
+}.freeze
+
+HALF_A_YEAR_SECONDS = (60 * 60 * 24) * (365.2425 / 2)
+
 def main
-  file_names = search_file_names
+  options = ARGV.getopts('alr')
+  searched_file_names = options['a'] ? Dir['..', '.*', '*'] : Dir['*']
+  sorted_file_names = searched_file_names.sort_by(&:downcase)
+  file_names = options['r'] ? sorted_file_names.reverse : sorted_file_names
+  options['l'] ? display_long_format(file_names) : display_column_format(file_names)
+end
+
+def display_long_format(file_names)
+  long_format_table = build_long_format_table(file_names)
+  total_blocks = long_format_table.sum { |format| format[:blocks] }
+  puts "total #{total_blocks}"
+
+  widths = create_long_format_widths(long_format_table)
+  long_format_table.each do |format|
+    rows = [
+      "#{format[:file_mode]} ",
```

Comment:
formatって形式とか書式みたいな意味なので、`format[:file_mode]`だと「file_modeの書式」みたいな意味に見えちゃいそうです。
それよりもファイルの情報やファイルのステータス、と捉えた方が意味が伝わりやすいかも？


```suggestion
  file_status_table.each do |file_status|
    rows = [
      "#{file_status[:file_mode]} ",
```